### DISCUSSION TITLE
[2024-01-08] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#8704] [Weekly Demo Day]
- James: Sort handling of Nothing within Column || and && #8656
- James: Prototype work for reading massive text files.
- Radosław: Finished: Unify `File_Format_Metadata` with `File_For_Read` #8628
- Radosław: work in progress on Enso Secret tests #8556 
- Greg: Support on_problems=Problem_Behavior.Report_Warning and Map_Error wrapping in Vector.map #8595
- Greg: Ignore User-Agent in HTTP tests #8630
- Jaroslav: Updating Espresso support & docs to match GraalVM for JDK21 #8641
- Jaroslav: Apply SectionsToBinOp pass to arguments of Section #8655
- Jaroslav: Read registered Persistance objects only once #8658
- Jaroslav: Avoid using cached bindings when the module isn't loaded from cache #8669
- Jaroslav: Canonicalize file names before comparing them #8670
- Jaroslav: Chained if_then_else application change #8671
- Jaroslav: Executing (parts of) Truffle TCK with Enso values #8685
   - Fix fix (`Character` to `Text`) delivered: https://github.com/enso-org/enso/pull/8685#issuecomment-1881310495
- Jaroslav: Replace Jackson library by simple data I/O streams #8693
- Hubert: Initialize builtin methods lazily in non-AOT mode #8654
- Hubert: Test benchmark theory #8668
- Hubert: Suppress logs for test cases unless a failure is reported #8694
- Hubert: Cats are not welcome #8696
- Hubert: Optimize AliasAnalysis hotspot #8701
- Pavel: Refactor Standard.Test to the builder API #8622
- Pavel: Remove org.bouncycastle dependency #8664
- Pavel: language server tests do not unnecessarily re-assemble fat jar #8695
- Dmitry: Unable to access constructor value in interactive mode #8626
- Dmitry: Kill lingering FanOutReceiver actors when the connection is closed #8642
- Dmitry: Lock critical section in projects/list command #8647
- Dmitry: Do not change project module name to Upper_Snake_Case #8697
- Ad: Make split to rows of Nothing value equal Nothing #8640 
- Ad: Give file read its own helper widget for delimiters. Remove newline add none. #8627

**Comment by JaroslavTulach** [comment]
The final results of last week speedups are visible in graph generated by:
```
enso/tools/performance/engine-benchmarks$ ./bench_download.py -s stdlib -b develop wip/jtulach/AvoidJacksonOnstartup_8553
```

![Empty startup speedups](https://github.com/enso-org/enso/assets/26887752/7cbe9ff2-f1c6-4ca7-9808-826fa1738e30)

On **Jan 2** the `Startup_empty_startup` benchmark was running in `3218ms`. On **Jan 8** it runs in `2794ms`. Result of:
- #8668
- #8664 
- #8696
- #8693 

E.g. **13% speedup**, congrats @Akirathan, @hubertp!
